test/verify/machineslib.py

```diff
@@ -1134,7 +1134,7 @@ def installWithErrorTest(dialog):
         # This is applicable only for the following test so let's keep it last,
         # in order to allow the rest of the tests to run faster with QEMU KVM
         # Run modprobe -r in retry loop because it fails sometimes with 'Module kvm is in use'
-        self.machine.execute("for i in 1 2 3 4 5; do modprobe -r kvm_intel && modprobe -r kvm && break || sleep 1; done")
+        wait(lambda: "Module KVM is in use" not in self.machine.execute("modprobe -r kvm_intel && modprobe -r kvm"))
```

Comment:
Does `modprobe -r` exit with zero on failure for you? It doesn't here (exits with 1), and also this string doesn't actually look like that: The module is spelled `kvm` (or `kvm_intel`):
```
❱❱❱ sudo modprobe -r kvm
modprobe: FATAL: Module kvm is in use.
❱❱❱ echo $?
1
```

So this condition will always be true. How did the previous version not work?

Reply:
It exits with 1 as well on failure of course. I will fix the typo. Anyway, I don't seem to reproduce locally anymore this issue with KVM. Not sure what was causing it in first place.

Reply:
Right. So I will tix the typo. Best would be to check the return mode in the lambda function instead of the output, but I don't seem to find example where we do it.

Reply:
You don't explicitly have to. `.execute()` throws an exception if the command fails (i. e. exits non-zero)